haystack/components/joiners/document_joiner.py

```diff
@@ -68,8 +95,15 @@ def __init__(
             If True sorts the Documents by score in descending order.
             If a Document has no score, it is handled as if its score is -infinity.
         """
-        if join_mode not in ["concatenate", "merge", "reciprocal_rank_fusion", "distribution_based_rank_fusion"]:
-            raise ValueError(f"DocumentJoiner component does not support '{join_mode}' join_mode.")
+        if isinstance(join_mode, str):
+            join_mode = JoinMode.from_str(join_mode)
```

Comment:
We need to override the `to_dict` and `from_dict` methods to correctly (de)serialize the enum. We should also add a couple of unit tests.

Reply:
Thanks for the review. Added in [ce7a196](https://github.com/deepset-ai/haystack/pull/8010/commits/ce7a196fbe57ad9db317ae7eb06df4e95103741c)